Pinta.Core/Managers/PaletteManager.cs

```
@@ -69,26 +69,35 @@ public Color SecondaryColor {
 		set => SetColor (false, value, true);
 	}
 
-	public Palette CurrentPalette { get; } = Palette.GetDefault ();
+	public Palette CurrentPalette { get; }
 
-	private readonly SettingsManager settings_manager;
-	private readonly PaletteFormatManager palette_format_manager;
+	private readonly ChromeManager chrome;
+	private readonly SettingsManager settings;
+	private readonly PaletteFormatManager palette_formats;
 	public PaletteManager (
-		SettingsManager settingsManager,
-		PaletteFormatManager paletteFormatManager)
+		ChromeManager chrome,
+		SettingsManager settings,
+		PaletteFormatManager paletteFormats)
 	{
 		List<Color> recentlyUsed = new (MAX_RECENT_COLORS);
 
 		recently_used = recentlyUsed;
 		RecentlyUsedColors = new ReadOnlyCollection<Color> (recentlyUsed);
 
-		settings_manager = settingsManager;
-		palette_format_manager = paletteFormatManager;
+		this.chrome = chrome;
+		this.settings = settings;
+		palette_formats = paletteFormats;
+
+		CurrentPalette = Palette.GetDefault ();
+
+		// This is an async method.
```

As an alternative idea, perhaps we could just not handle the exception here and let it propagate back to the generic exception handler
In this case we're loading the saved palette from Pinta's preferences, so it would be very unexpected for there to be errors when loading it unless something was corrupted

---

That is a good idea @cameronwhite, and we wouldn't have to worry about it being an `async` method either. I just made the change.

---

yeah, the generic exception handler is initialized very early on:
https://github.com/PintaProject/Pinta/blob/68355d6117c457ad0a6b3faeda74cd7ba9920463/Pinta/Main.cs#L82